farm-staking/farm-staking/src/lib.rs

```diff
@@ -144,4 +149,34 @@ pub trait FarmStaking:
             "May only call this function through VM query"
         );
     }
+
+    fn try_set_farm_position_migration_nonce(&self) {
+        if !self.farm_position_migration_nonce().is_empty() {
+            return;
+        }
+
+        let farm_token_mapper = self.farm_token();
+
+        let attributes = StakingFarmTokenAttributes {
+            reward_per_share: BigUint::zero(),
+            compounded_reward: BigUint::zero(),
+            current_farm_amount: BigUint::zero(),
+            original_owner: self.blockchain().get_sc_address(),
+        };
+
+        let migration_farm_token_nonce = if farm_token_mapper.get_token_state().is_set() {
```

Comment:
Same as above - use GetCurrentESDTNFTNonce. Why do you have duplicated code ? Base function should resolv this - as code is exactly the same.

Reply:
Updated.

Reply:
The issue for the code duplication was `farm_token`. It has different attributes for farm and farm-staking.
I update the code: moved this to common module and provide the token_mapper as a parameter.